frontend/src/lib/api.ts

```diff
@@ -101,7 +101,9 @@ export function useSaveScratch(localScratch: Scratch): () => Promise<Scratch> {
 }
 
 export async function claimScratch(scratch: Scratch): Promise<void> {
-    const { success } = await post(`${scratchUrl(scratch)}/claim`, {})
+    const { success } = await post(`${scratchUrl(scratch)}/claim`, {
+        token: scratch.claim_token,
```

Comment:
Correct me if I'm wrong but it seems this will be `undefined` when scratch didn't come from create_scratch, e.g. if you just visit an unclaimed scratch

Reply:
If https://github.com/decompme/decomp.me/pull/945#discussion_r1454839593 is applied, then throw an error if the claim token is missing on `scratch`

Reply:
Updated TS types for this